docs/README_Cross-Distro_bootstrap.md

```diff
@@ -0,0 +1,211 @@
+# Introduction
+
+The AMIgen utilities are typically used to generate AMIs from a starting-point (or, "bootstrap") AMIs. The starting-point AMIs may be selected from the AWS Marketplace (or CSP's equivalent), public AMIs or private AMIs. These starting-point AMIs must be the same distribution as the creation-target for the AMIgen scripts (e.g., use a Red Hat Enterprise Linux starting-point to create a Red Hat Enterprise Linux final AMI; use a Alma Linux starting-point to create a Alma Linux final AMI; etc.).
+
+# Usage
+
+Depending on your CSP's environment, there may be no suitable starting-point AMIs to use. If such is the case, use a procedure similar to the following to create such starting-point AMIs.
+
+## Execute a `chroot()`-style build
+
+1. Launch an EL8 AMI (probably RHEL8), ensuring that a secondary disk of at least 8GiB in size is attached
+1. Install the `@development` package-group to the resultant EC2
+1. Login to the EC2
+1. Change to the `root` user (e.g., `sudo -i`)
+1. Clone the AMIgen8 project (this project) to the `root` user's `${HOME}`
+1. Use the `XdistroSetup.sh` script to stage the necessary alternate-disto GPG and repository files to the build-environment:
+    ~~~
+    ./XdistroSetup.sh -d <DISTRO_NAME> \
+      -k <URL_TO_GPG_KEYFILE_1>,<URL_TO_GPG_KEYFILE_2>,...,<URL_TO_GPG_KEYFILE_n>, \
+      -r <URL_TO_DISTRO_RELEASE_FILE_1>,<URL_TO_DISTRO_RELEASE_FILE_2>,...,<URL_TO_DISTRO_RELEASE_FILE_n>,
+    ~~~
+    For Rocky Linux 8, this would look something like:
+    ~~~
+    ./XdistroSetup.sh -d RockyLinux \
+      -k https://download.rockylinux.org/pub/rocky/RPM-GPG-KEY-rockyofficial \
+      -r https://download.rockylinux.org/pub/rocky/8/BaseOS/x86_64/os/Packages/r/rocky-release-8.5-2.el8.noarch.rpm,https://download.rockylinux.org/pub/rocky/8/BaseOS/x86_64/os/Packages/r/rocky-repos-8.5-2.el8.noarch.rpm
+    ~~~
+1. Navigate into the AMIgen8 project-root (e.g., `cd AMIgen8`)
+1. (Optional) Clean up target-disk by executing:
+    ~~~
+    ./Umount.sh -C <TARGET_DEV>
+    ~~~
+1. Partition the target-disk by executing:
+    ~~~
+    ./DiskSetup.sh -B 1024 -d <TARGET_DEV> -f xfs -r <FILESYSTEM_LABEL>
+    ~~~
+    Note: Selection of EXTn filesystem-types is also supported. Change the above `xfs`  and all following `xfs` references, as necessary, if an EXTn filesystem-type is chosen, instead.
+1. Mount the target-disk by executing:
+    ~~~
+    ./MkChrootTree.sh -d <TARGET_DEV> -f xfs -m /mnt/ec2-root --no-lvm
+    ~~~
+1. Install the OS packages by executing:
+    ~~~
+    ./OSpackages.sh -a <LIST_OF_CHANNEL_NAMES> \
+      -m /mnt/ec2-root -r /root/RPM/<DISTRO_NAME>/<DISTRO_RELEASE_RPM> -X 
```

Comment:
This would work just fine without staging the rpms first... just use the urls to the rpms...

```suggestion
      -m /mnt/ec2-root -r https://url/to/repo.rpm,https://url/to/release.rpm -X 
```

Reply:
Note added.